### ISSUE TITLE
Add dateStarted and dateCompleted to JSON schema in the right place

### ISSUE THREAD

**Issue by rdb** [bible-technology/scripture-burrito#89]
Currently, the dateStarted and dateCompleted fields are placed in the identification section.  It appears that this was inherited from the current DBL schema.  However, it does not seem like it is a major identifying property of a burrito.  In fact, Paratext does not care about these fields to my knowledge, and the Registry does not even consider these metadata but aspects of project progress, which (IIRC) was considered to be out-of-scope for Scripture Burrito.

Should we consider whether these fields are better placed elsewhere, or needed at all?  Perhaps downstream consumers of metadata could comment on the use of these fields?

**Comment by jag3773** [comment]
I agree that those probably shouldn't be in the identification section.

Project progress is intended to be included in SB but I don't think we have specified that yet.  I think it would be most natural for that information to be included in that forthcoming section.

**Comment by mvahowe** [comment]
We could make that section and put two fields in it to be going on with?

**Comment by rdb** [comment]
I'm very interested in learning about the use cases for storing progress information in burritos.  I vaguely recall this was present in an early draft but scrapped after it was thought to be out-of-scope.  In our ecosystem, DBL has no interest in it as we have an entirely different pipeline for progress information.

**Comment by mvahowe** [comment]
This is one of these circular conversations, because we never get to the end of defining scope. In the current workflows, all the fun stuff happens within PTX, which sometimes spits out a DBL bundle once all the work has been done. Recent comments have suggested that we're considering projects that move between tools during translation and, in that case, we'd surely need a way to preserve or recover progress information across those tools. (I still think this implies hooks to ecosystems via servers rather than including all the social information in burritos, but hooks still need to be represented.)

In other words, is SB is an "on the way to publication" export format (which, essentially, described DBL bundles), we don't need progress information (although we do need to support burritos destined for incremental publishing.) But my understanding is that there is a desire to use SB as a way to move data during translation and, in that case, publication plans would surely be in scope (or, at the very least, in the immediate context.)

The progress features we scrapped were 2 parallel attempts to represent current PTX progress in DBL. Whatever we need in SB in the future, that wasn't it.

**Comment by rdb** [comment]
Thank you for clarifying this aspect of SB usage.  In this case, I can understand that progress (and assignments, planning, etc.) will need to be represented, either in the metadata or in a separate file in the burrito (though I expect that creating a universal standard for project planning will be a challenging exercise).

**Comment by mvahowe** [comment]
Right. The hard conversation we keep avoiding is about things that would be challenging to implement (technically and/or politically) vs things we won't be able to do if we don't implement those challenging features.

**Comment by rdb** [comment]
I slightly changed this issue to be a task to add this to the JSON schema.

**Comment by jag3773** [comment]
We decided to wait on this but need to address before 1.0.

**Comment by mvahowe** [comment]
I think that "waiting on this" should mean that we keep these two fields, since otherwise DBL will have to throw that data away. I'm inclined to leave them in identification for now. (I agree it's an odd place for them to be, but I don't think we've agreed enough on other issues to decide if the new section should be called history or progress or targets or something else.)

If we can agree on a plan I can roll this into my next PR.

**Comment by jag3773** [comment]
In the interim let's put them into a `progress` section and we can expand or modify as needed before 1.0.

**Comment by klassenjm** [comment]
Mark thought I might take a peek at this issue.

I'm not aware of DBL metadata ever having a `dateStarted` element. But as far as the `dateCompleted` element is concerned - I think this was likely in `identification` because it is a field which in some manner is distinguishing one edition of a translation from another one. For example referring to the 2004 edition of the New Living Translation and not the 2007 one. That may not be the only pratical use case. Or simply helping a person looking at a list of translations in a particular language to distinguish one from another.

I don't see a problem with it sitting in `progress`

**Comment by mvahowe** [comment]
So:
```
"progress": {
    "dateCompleted": "2017-12-01"
}
```
where
* "progress" is optional
* "dateCompleted" is required if "progress" is present
* the timestamp is YYYY-MM-DD or YYYY-MM or YYYY

**Comment by mvahowe** [comment]
Plan B:
```
"progress": {
    "dateStarted": "2017-11-30",
    "dateCompleted": "2017-12-01"
}
```
where
* "progress" is optional
* at least one of "dateStarted" and "dateCompleted" is required if "progress" is present
* the timestamp is YYYY-MM-DD or YYYY-MM or YYYY
